drivers/vulkan/vulkan_context.cpp

```diff
@@ -1960,18 +1977,13 @@ Error VulkanContext::_update_swap_chain(Window *window) {
 
 	free(presentModes);
 
-	// Determine the number of VkImages to use in the swap chain.
-	// Application desires to acquire 3 images at a time for triple
-	// buffering.
-	uint32_t desiredNumOfSwapchainImages = 3;
-	if (desiredNumOfSwapchainImages < surfCapabilities.minImageCount) {
-		desiredNumOfSwapchainImages = surfCapabilities.minImageCount;
-	}
+	uint32_t desiredNumOfSwapchainImages = MAX(surfCapabilities.minImageCount, swapchain_desired_count);
```

Comment:
Some sources I found seem to state it's sensible to also lower bound this by `surfCapabilities.minImageCount+1` (this might, on particularly conservative implementations, relax the contention for driver resources a bit when acquiring the next image; the common case here is still that the default 3 images are the value that you are gonna get)

Reply:
I found all references to this info trace back to the [Vulkan Tutorial](https://vulkan-tutorial.com/Drawing_a_triangle/Presentation/Swap_chain).

After research I found the following:

1. The OSes that have sync issues (like X11 on Linux) already set minImageCount to a large enough value like 4 or 5.
2. I suspect it is highly likely that the tutorial confused "wait on the driver to complete internal operations" with how VSync is supposed to work. This is known and can be observed in [my VSync Simulator](https://darksylinc.github.io/vsync_simulator/). If you set swapchain_count to 2 and the rendering time is higher than the refresh rate, then the entire framerate will alternate to an integer divisor of the refresh rate (e.g. 60fps, 30 fps, 15 fps). Depending on how the OS compositor internally works, this might happen with swapchain_count > 2.
    - Seeing the [other ticket](https://github.com/godotengine/godot/issues/71795), it is clear some users explicitly want this behavior (because if the CPU & GPU are fast enough to not miss any VBLANK, then latency is minimum).